packages/@dcl/sdk-commands/src/commands/start/data-layer/rpc-engine.ts

```diff
@@ -0,0 +1,99 @@
+import * as components from '@dcl/ecs/dist/cjs/components'
+import { Schemas, IEngine, Engine, Entity, DeepReadonly, CrdtMessageType } from '@dcl/ecs'
+import { ByteBuffer, ReadWriteByteBuffer } from '@dcl/ecs/dist/cjs/serialization/ByteBuffer'
+import { CliComponents } from '../../../components'
+import { resolve } from 'path'
+
+type ComponentId = number
+type EngineState = Record<number, Record<ComponentId, DeepReadonly<unknown>>>
+
+function getEngineState(engine: IEngine): EngineState {
+  const entities: EngineState = {}
+
+  function ensureEntityExists(entity: Entity) {
+    if (!entities[entity]) {
+      entities[entity] = {}
+    }
+    return entities[entity]
+  }
+
+  for (const component of engine.componentsIter()) {
+    for (const [entity, componentValue] of engine.getEntitiesWith(component)) {
+      const data = ensureEntityExists(entity)
+      data[component.componentId] = componentValue
+    }
+  }
+
+  return entities
+}
+
+export async function saveEngineState(components: Pick<CliComponents, 'fs'>, path: string, engine: IEngine) {
+  const engineState = getEngineState(engine)
+  await components.fs.writeFile(resolve(path, 'engine-state.json'), JSON.stringify(engineState, null, 2))
+}
+
+/**
+ * Serialize engine and send the Messages to the inspector/babylon engine
+ */
+export function serializeEngine(engine: IEngine) {
+  const messages: ByteBuffer = new ReadWriteByteBuffer()
+
+  // TODO: add deleted entities messages
+  // add component values
+  for (const component of engine.componentsIter()) {
+    component.dumpCrdtState(messages)
+  }
+  return messages.toBinary()
+}
+
+function createEditorComponents(engine: IEngine) {
+  const Label = engine.defineComponent('inspector::Label', {
+    label: Schemas.String
+  })
+
+  const EntitySelected = engine.defineComponent('editor::EntitySelected', {
+    gizmo: Schemas.Int
+  })
+
+  const Toggle = engine.defineComponent('inspector::Toggle', {})
+  const Transform = components.Transform(engine as any)
+  const MeshRenderer = components.MeshRenderer(engine as any)
+  return { Label, EntitySelected, Toggle, Transform, MeshRenderer }
+}
```

Comment:
this function MUST to be shared between inspector and data-layer. otherwise a change in a schema will break everything